### PR TITLE
Use paginated approach for order fetching

### LINKED ISSUE #1: Python scripts fail, 'Out of gas'
Running:
```
python3 dex-visualization-tools/generate_plot_for_all_pairs.py DAI OWL
```
fails returning a traceback (see below). All token pairs are affected.

Traceback:
```
(venv) $ python3 plot_orderbook_tokenpair.py DAI OWL
Traceback (most recent call last):
  File "plot_orderbook_tokenpair.py", line 173, in <module>
    inst = util.read_instance_from_blockchain()
  File "/path/dex-visualization-tools/util.py", line 106, in read_instance_from_blockchain
    orders = contract_reader.get_orderbook()
  File "/path/dex-visualization-tools/contract_reader.py", line 82, in get_orderbook
    orders_encoded = contract.functions.getEncodedOrders().call()
  File "/path/dex-visualization-tools/venv/lib/python3.7/site-packages/web3/contract.py", line 904, in call
    **self.kwargs
  File "/path/dex-visualization-tools/venv/lib/python3.7/site-packages/web3/contract.py", line 1424, in call_contract_function
    return_data = web3.eth.call(call_transaction, block_identifier=block_id)
  File "/path/dex-visualization-tools/venv/lib/python3.7/site-packages/eth_utils/functional.py", line 45, in inner
    return callback(fn(*args, **kwargs))
  File "/path/dex-visualization-tools/venv/lib/python3.7/site-packages/web3/eth.py", line 426, in call
    [transaction, block_identifier],
  File "/path/dex-visualization-tools/venv/lib/python3.7/site-packages/web3/manager.py", line 152, in request_blocking
    raise ValueError(response["error"])
ValueError: {'code': -32015, 'message': 'VM execution error.', 'data': 'Out of gas'}
```

### PR DESCRIPTION
**Pull request by josojo**
closes #1 

@twalth3r  started this PR and I finished it.

testplan:

```
docker build -t order_book_printer .
docker run -it -v ~/PATH_TO_FOLDER/dex-visualization-tools/:/app/ order_book_printer
```

Inside the docker, you can visualize the orderbook:

```
python3 plot_orderbook_tokenpair.py PAX WETH
```

Observe that it is working as before

### CODE REVIEW

File: contract_reader.py [hunk 1]
```diff
@@ -115,11 +104,62 @@ def get_orderbook():
...
+    Returns:
+        A list of all orders.
+
+    """
+    orders = get_orderbook()
+
+    # Skip orders that are no longer or not yet valid.
+    batch_id = get_current_batch_id()
+    orders = list(filter(lambda order: order['validUntil'] <
+                         batch_id or order['validFrom'] > batch_id, orders))
```
Comment (marcovc) [inline contract_reader.py]: 1. Not sure I understand what we want here, but If `orders` is the set of orders that include the current batch_id, then the condition should be `validUntil>batchId and validFrom<batchId. 

2. I never know what the convention is, but I find it difficult to believe that this interval is open on both sides. It should be closed on one of them, right?

File: contract_reader.py [hunk 2]
```diff
@@ -115,11 +104,62 @@ def get_orderbook():
...
+            contract.functions.getEncodedUsersPaginated(
+                Web3.toChecksumAddress(current_user),
+                current_offset, page_size).call())
+
+        orders += orders_decoded_page
+        if len(orders_decoded_page) > 0:
+            current_user = orders_decoded_page[-1]['accountID']
+            current_offset = len(
+                [o for o in orders if o['accountID'] == current_user])
+
```
Comment (marcovc) [inline contract_reader.py]: Isn't `current_offset = len(orders)` ?

**Review by marcovc — APPROVED** [review]
Thanks!

File: contract_reader.py (hunk 2 above)
Comment (josojo) [reply contract_reader.py]: No, it is always an offset per user, due to the way, we are storing the orders per user in the smart contract

File: contract_reader.py (hunk 1 above)
Comment (josojo) [reply contract_reader.py]: I just copy-pasted the line from the current python script. But in there,  it used the inverse logic. Thanks for highlighting!

File: contract_reader.py [hunk 3]
```diff
@@ -105,7 +93,8 @@ def get_orderbook():
             continue
```
Comment (twalth3r) [inline contract_reader.py]: @josojo - there is another filtering going on here which would need to be removed!

**Review by twalth3r — APPROVED** [review]

**Comment by josojo** [comment]
Thank you all

**Pull request merged**